src/Custom/Assistants/AssistantClient.cs

```diff
@@ -691,77 +687,107 @@ public virtual ClientResult<bool> DeleteMessage(string threadId, string messageI
+            return operation;
+        }
+
+        await operation.WaitAsync(cancellationToken);
+        return operation;
     }
 
-    /// <summary>
-    /// Begins a new <see cref="ThreadRun"/> that evaluates a <see cref="AssistantThread"/> using a specified
-    /// <see cref="Assistant"/>.
-    /// </summary>
-    /// <param name="threadId"> The ID of the thread that the run should evaluate. </param>
-    /// <param name="assistantId"> The ID of the assistant that should be used when evaluating the thread. </param>
-    /// <param name="options"> Additional options for the run. </param>
-    /// <param name="cancellationToken">A token that can be used to cancel this method call.</param>
-    /// <returns> A new <see cref="ThreadRun"/> instance. </returns>
-    public virtual ClientResult<ThreadRun> CreateRun(string threadId, string assistantId, RunCreationOptions options = null, CancellationToken cancellationToken = default)
+    public virtual RunOperation CreateRun(
+        ReturnWhen returnWhen,
+        string threadId,
+        string assistantId,
+        RunCreationOptions options = null,
+        CancellationToken cancellationToken = default)
     {
         Argument.AssertNotNullOrEmpty(threadId, nameof(threadId));
         Argument.AssertNotNullOrEmpty(assistantId, nameof(assistantId));
+
         options ??= new();
         options.AssistantId = assistantId;
         options.Stream = null;
 
+
         ClientResult protocolResult = CreateRun(threadId, options.ToBinaryContent(), cancellationToken.ToRequestOptions());
-        return CreateResultFromProtocol(protocolResult, ThreadRun.FromResponse);
+        ClientResult<ThreadRun> result = CreateResultFromProtocol(protocolResult, ThreadRun.FromResponse);
+
+        RunOperation operation = new RunOperation(_pipeline, _endpoint,
+            value: result,
+            status: result.Value.Status,
+            cancellationToken.ToRequestOptions(),
+            result.GetRawResponse());
+
+        if (returnWhen == ReturnWhen.Started)
+        {
+            return operation;
+        }
+
+        operation.Wait(cancellationToken);
+        return operation;
     }
 
-    /// <summary>
-    /// Begins a new streaming <see cref="ThreadRun"/> that evaluates a <see cref="AssistantThread"/> using a specified
-    /// <see cref="Assistant"/>.
-    /// </summary>
-    /// <param name="threadId"> The ID of the thread that the run should evaluate. </param>
-    /// <param name="assistantId"> The ID of the assistant that should be used when evaluating the thread. </param>
-    /// <param name="options"> Additional options for the run. </param>
-    /// <param name="cancellationToken">A token that can be used to cancel this method call.</param>
-    public virtual AsyncCollectionResult<StreamingUpdate> CreateRunStreamingAsync(
-        string threadId,
-        string assistantId,
-        RunCreationOptions options = null,
+    public virtual async Task<RunOperation> ContinueRunAsync(
```

Comment:
Instead of adding a new method group here, add a factory method to RunOperation to rehydrate it, similar to Azure.Core Operation static Rehydrate method.